django/contrib/admin/static/admin/js/selectfilter.js

```diff
@@ -0,0 +1,345 @@
+            selector.find('.selector-remove').toggleClass('active', is_to_selected);
+            // Active if the corresponding box isn't empty
+            selector.find('.selector-chooseall').toggleClass('active', available.find('option').length > 0);
+            selector.find('.selector-clearall').toggleClass('active', chosen.find('option').length > 0);
+        },
+        refresh_state: function(selector) {
+            settings.refresh_icons(selector);
+            var available = selector.data('available');
+            var chosen = selector.data('chosen');
+
+            // Re-generate the content of the actual, hidden, box.
+            var actualBox = selector.data('actualBox');
+            var actualOptions = actualBox.children();
+
+            // De-select all options.
+            available.children().each(function(){
+                var option = actualOptions[$(this).attr('data-index')];
+                option.selected = false;
+            });
+
+            // Select all the same options as those from the 'to' box.
+            chosen.children().each(function(){
+                var option = actualOptions[$(this).attr('data-index')];
+                option.selected = true;
+            });
+        },
+        move: function(from, to, all) {
+            var selector = from.data('selector');
+            var available = selector.data('available');
+
+            var options;
+            if (all) {
+                options = from.children();
+            }
+            else {
+                options = from.children().filter(':selected');
+            }
+
+            if (from == available) {
+                options.each(function() {
+                    var moved_option = $(this);
+                    available.data('cache').each(function(index, option) {
+                        var cached_option = $(option);
+                        if (moved_option.val() == cached_option.val()) {
+                            available.data('cache').splice(index, 1);
+                        }
+                    });
+                });
+            }
+            else {
+                options.each(function() {
+                    available.data('cache').push($(this).clone()[0]);
+                });
+            }
+
+            to.append(options);
+            to.children().removeAttr('selected');
+            settings.refresh_state(from.data('selector'));
+        }
+    };
+
+    var methods = {
+        init: function(options) {
+            $.extend(settings, options);
+
+            var actualBox = this;
+            var stack_class = settings.stacked ? 'selector stacked' : 'selector';
+            var selector = $('<div>').addClass(stack_class)
+                                     .attr('id', actualBox.attr('id') + '_selector')
+                                     .appendTo(actualBox.parent());
+
+            actualBox.parent().find('p').each(function(){
+                if ($(this).hasClass('info')) {
```

Comment:
Hoist `$(this)` into a local variable, to avoid reconverting it again inside both conditionals and their branches?